app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/NewTransferViewModel.kt

```diff
@@ -129,12 +143,110 @@ class NewTransferViewModel @Inject constructor(
 
     fun selectTransferType(type: TransferType) {
         _selectedTransferType.value = type
+        savedStateHandle[SELECTED_TRANSFER_TYPE] = type
+    }
+    //endregion
+
+    //region Transfer Advanced Options
+
+    private val _selectedValidityPeriodOption = MutableStateFlow<ValidityPeriodOption?>(
+        savedStateHandle[SELECTED_VALIDITY_PERIOD_KEY]
+    )
+    val selectedValidityPeriodOption: StateFlow<ValidityPeriodOption?> = _selectedValidityPeriodOption.asStateFlow()
+
+    private val _selectedDownloadLimitOption = MutableStateFlow<DownloadLimitOption?>(
+        savedStateHandle[SELECTED_DOWNLOAD_LIMIT_KEY]
+    )
+    val selectedDownloadLimitOption: StateFlow<DownloadLimitOption?> = _selectedDownloadLimitOption.asStateFlow()
+
+    private val _selectedPasswordOption = MutableStateFlow<PasswordTransferOption?>(
+        savedStateHandle[SELECTED_PASSWORD_OPTION_KEY] ?: PasswordTransferOption.NONE
+    )
+    val selectedPasswordOption: StateFlow<PasswordTransferOption?> = _selectedPasswordOption.asStateFlow()
+
+    private val _selectedLanguageOption = MutableStateFlow<EmailLanguageOption?>(savedStateHandle[SELECTED_LANGUAGE_KEY])
+    val selectedLanguageOption: StateFlow<EmailLanguageOption?> = _selectedLanguageOption.asStateFlow()
+
+    private fun selectTransferValidityPeriod(validityPeriodOption: ValidityPeriodOption) {
+        _selectedValidityPeriodOption.value = validityPeriodOption
+        savedStateHandle[SELECTED_VALIDITY_PERIOD_KEY] = validityPeriodOption
+    }
+
+    private fun selectTransferDownloadLimit(downloadLimit: DownloadLimitOption) {
+        _selectedDownloadLimitOption.value = downloadLimit
+        savedStateHandle[SELECTED_DOWNLOAD_LIMIT_KEY] = downloadLimit
+    }
+
+    private fun selectTransferPasswordOption(passwordOption: PasswordTransferOption) {
+        _selectedPasswordOption.value = passwordOption
+        savedStateHandle[SELECTED_PASSWORD_OPTION_KEY] = passwordOption
+    }
+
+    private fun selectTransferLanguage(language: EmailLanguageOption) {
+        _selectedLanguageOption.value = language
+        savedStateHandle[SELECTED_LANGUAGE_KEY] = language
+    }
+
+    fun initTransferAdvancedOptionsValues() {
+        viewModelScope.launch(ioDispatcher) {
+            appSettingsManager.getAppSettings()?.let {
+                selectTransferValidityPeriod(it.validityPeriod.toAdvancedOption())
+                selectTransferDownloadLimit(it.downloadLimit.toAdvancedOption())
+                selectTransferLanguage(it.emailLanguage.toAdvancedOption())
+            } ?: run {
+                SentryLog.e(TAG, "AppSettings is null in ImportFilesScreen, it should not happened")
+                selectTransferValidityPeriod(ValidityPeriodOption.THIRTY)
+                selectTransferDownloadLimit(DownloadLimitOption.TWO_HUNDRED_FIFTY)
+                selectTransferLanguage(EmailLanguageOption.FRENCH)
+            }
+        }
+    }
```

Comment:
I was discussing this with @sirambd, but maybe there's an even better solution than using LaunchedEffects or ViewMode.init(). I feel like it might be applicable here maybe? This is still kinda uncertain maybe?

Anyway, the idea is to prefer cold flows that kick off their work when you start collecting them via collectAsStateWithLifecycle()

Source: https://proandroiddev.com/loading-initial-data-in-launchedeffect-vs-viewmodel-f1747c20ce62

Reply:
It's interesting, we could do this later